deck_chores/config.py

```diff
@@ -73,6 +75,19 @@ def generate_config() -> None:
         environment=local_environment,
     )
 
+    tmp_job_name_regex = getenv('JOB_NAME_REGEX', '[a-z0-9-]+')
+
+    try:
+        re.compile(tmp_job_name_regex)
+        is_valid = True
+    except re.error:
+        is_valid = False
+
+    if not is_valid:
+        tmp_job_name_regex = '[a-z0-9-]+'
+    
+    cfg.job_name_regex = fr'{tmp_job_name_regex}'
+
```

Comment:
1) please let an invalid regex terminate the program with an error.
2) it'd be preferable that temporary variables were avoided.
3) a regex containing literal dots would be invalid.